### PR TITLE
[#2727] Improve Sandbox deployment scripts.

### PR DESCRIPTION
**Pull request by b-abel**
This moves the commands for creating a certificate into another script than the one for installing cert-manager. While the cert-manager installation should be performed only once, requesting certificates may happen more often. The commit contains other small improvements to the sandbox deployment.

### CODE REVIEW

**Comment by b-abel** [comment]
@sophokles73 Would you mind taking a look?

File: deploy/src/main/sandbox-k8s/10-deploy-cert-manager.sh [hunk 1]
```diff
@@ -12,43 +12,20 @@
...
+helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --set installCRDs=true --wait --timeout 5m0s
 
-# checking when cert-manager is ready is tricky: https://cert-manager.io/docs/installation/kubernetes/#verifying-the-installation
-# we just retry the following command
 echo ""
-echo "Adding issuer Let's Encrypt to cert-manager with API: $LE_API"
-LE_CONFIG=$(helm template ./letsencrypt-chart -f ./letsencrypt-chart/values-$LE_API.yaml --set spec.acme.email=$EMAIL)
-retry echo "$LE_CONFIG" | kubectl apply -f - $KUBECONFIG || true # do not abort although '-e' is set
+echo "Creating namespace 'hono'..."
+kubectl create namespace hono
```
Comment (sophokles73) [inline deploy/src/main/sandbox-k8s/10-deploy-cert-manager.sh:31]: why do we create the *hono* namespace here? This doesn't seem to be related to the cert-manager, is it?

Comment (b-abel) [reply deploy/src/main/sandbox-k8s/10-deploy-cert-manager.sh:31]: Right, it seems a little misplaced. Therefore it is explicitly mentioned in the comments of the script and in the readme. The point is that the following script needs the namespace to create the secret for the certificate in it. Having an additional script only for creating the namespace seemed a little bit like overkill to me. But maybe that would be cleaner...

Comment (sophokles73) [reply deploy/src/main/sandbox-k8s/10-deploy-cert-manager.sh:31]: why don't we simply mention it in the readme that you need to create the namespace using `kubectl create namespace hono`?

Comment (b-abel) [reply deploy/src/main/sandbox-k8s/10-deploy-cert-manager.sh]: Done.

File: deploy/src/main/sandbox-k8s/30-deploy-hono.sh [hunk 2]
```diff
@@ -12,17 +12,18 @@
...
+echo ""
+echo "Adding the Helm repository for Hono..."
+helm repo add eclipse-iot https://eclipse.org/packages/charts
+helm repo update
 
 echo ""
-echo "Deploying hono..."
-helm install eclipse-hono --dependency-update -f hono-values.yml -n hono eclipse-iot/hono $KUBECONFIG $HELM_WAIT
+echo "Deploying Hono..."
+helm install eclipse-hono --dependency-update -f $SCRIPTPATH/hono-values.yml -n hono eclipse-iot/hono --wait --timeout 5m0s
```
Comment (sophokles73) [inline deploy/src/main/sandbox-k8s/30-deploy-hono.sh:29]: why don't we use the `--create-namespace` and `--namespace` options here as well (as we do when installing cert-manager)? We do not seem to be deploying the certificates to namespace *hono*, though, are we?

Comment (b-abel) [reply deploy/src/main/sandbox-k8s/30-deploy-hono.sh:29]: The certificates are indeed put into a secret in the namespace _hono_. You can find this configuration in the file `letsencrypt-certificate.yml`.

File: deploy/src/main/sandbox-k8s/readme.md [hunk 3]
```diff
@@ -9,17 +9,19 @@ This folder contains scripts and configuration files for setting up a Hono *sand
...
-execute the script again with `production` as the second argument to request a valid certificate from the productive API.
-
-The scripts need to be executed with `sudo` because the Kube config is only readable by root.
+The scripts need to be executed in the order of the numbers in the file names.
+
+* Install Kubernetes with: `./00-install-k3s.sh`. Export the Kubernetes config as an environment variable, 
+   e.g., like: `export KUBECONFIG=/etc/rancher/k3s/k3s.yaml`. 
+* Deploy the [cert-manager](https://cert-manager.io/): `./10-deploy-cert-manager.sh`.
+* Create the namespace "hono": `kubectl create namespace hono`.
+* Request a public certificate from Let's Encrypt: `./20-create-certificate.sh $LE_EMAIL`. Verify that the certificate
```
Comment (sophokles73) [inline deploy/src/main/sandbox-k8s/readme.md:18]: `$LE_EMAIL` ?

File: deploy/src/main/sandbox-k8s/30-deploy-hono.sh (hunk 2 above)
Comment (sophokles73) [reply deploy/src/main/sandbox-k8s/30-deploy-hono.sh:29]: I see

File: deploy/src/main/sandbox-k8s/readme.md (hunk 3 above)
Comment (b-abel) [reply deploy/src/main/sandbox-k8s/readme.md]: Better now?

File: deploy/src/main/sandbox-k8s/readme.md [hunk 4]
```diff
@@ -9,17 +9,21 @@ This folder contains scripts and configuration files for setting up a Hono *sand
...
+The scripts need to be executed in the order of the numbers in the file names.
+
+* Install Kubernetes with: `./00-install-k3s.sh`. Export the Kubernetes config as an environment variable, 
+   e.g., like: `export KUBECONFIG=/etc/rancher/k3s/k3s.yaml`. 
+* Deploy the [cert-manager](https://cert-manager.io/): `./10-deploy-cert-manager.sh`.
+* Create the namespace "hono": `kubectl create namespace hono`.
+* Requesting a public certificate from Let's Encrypt requires an email address that Let's Encrypt uses to contact you
+  Please make sure to replace "<your-email-address>" in the following commands by an email address that is actually used.
+  Request the certificate with: `./20-create-certificate.sh <your-email-address>`. Verify that the certificate
+  has successfully been issued with `kubectl describe certificates hono-eclipseprojects-io`. When this looks good,
```
Comment (sophokles73) [inline deploy/src/main/sandbox-k8s/readme.md:21]: this will probably not work if the certificates have been deployed to namespace *hono*, will it?

File: deploy/src/main/sandbox-k8s/readme.md (hunk 3 above)
Comment (sophokles73) [reply deploy/src/main/sandbox-k8s/readme.md]: yes

File: deploy/src/main/sandbox-k8s/readme.md (hunk 4 above)
Comment (b-abel) [reply deploy/src/main/sandbox-k8s/readme.md]: Right. My bad. I added the namespace.

**Review by sophokles73 — APPROVED** [review]
LGTM
